presentation/src/main/java/team/jsv/icec/ui/main/mosaic/mosaicFace/MosaicFaceFragment.kt

```diff
@@ -1,57 +1,75 @@
 package team.jsv.icec.ui.main.mosaic.mosaicFace
 
 import androidx.fragment.app.activityViewModels
+import androidx.lifecycle.flowWithLifecycle
 import androidx.lifecycle.lifecycleScope
 import com.google.android.material.slider.Slider
 import dagger.hilt.android.AndroidEntryPoint
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.Job
-import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
+import team.jsv.domain.model.MosaicType
 import team.jsv.icec.base.BaseFragment
 import team.jsv.icec.base.EventObserver
 import team.jsv.icec.ui.main.mosaic.MosaicViewModel
 import team.jsv.presentation.R
 import team.jsv.presentation.databinding.FragmentMosaicFaceBinding
 
 @AndroidEntryPoint
-class MosaicFaceFragment :
-    BaseFragment<FragmentMosaicFaceBinding>(R.layout.fragment_mosaic_face) {
+class MosaicFaceFragment
+    : BaseFragment<FragmentMosaicFaceBinding>(R.layout.fragment_mosaic_face) {
 
     private val viewModel: MosaicViewModel by activityViewModels()
 
-    private var debounceJobGetMosaicImage: Job? = null
-    private val debounceDelay: Long = 300L
-
     override fun initView() {
-        viewModel.backPress.observe(this, EventObserver {
-            popBackStack()
-        })
+        initClickListener()
+        observeState()
+    }
 
-        viewModel.pixelSize.observe(this) {
-            binding.sliderMosaicFigure.value = it
+    private fun initClickListener() {
+        binding.ivMosaicFigure.setOnClickListener {
+            viewModel.setMosaicType(MosaicType.Mosaic)
+        }
+        binding.ivBlurFigure.setOnClickListener {
+            viewModel.setMosaicType(MosaicType.Blur)
+        }
+        binding.ivRefresh.setOnClickListener {
+            viewModel.mosaicFaceRefresh()
         }
-
         binding.sliderMosaicFigure.apply {
-            haloRadius = 0
-
             addOnChangeListener(Slider.OnChangeListener { _, value, _ ->
                 viewModel.setPixelSize(value)
             })
 
             addOnSliderTouchListener(object : Slider.OnSliderTouchListener {
                 override fun onStartTrackingTouch(slider: Slider) {}
                 override fun onStopTrackingTouch(slider: Slider) {
-                    debounceJobGetMosaicImage?.cancel()
-                    debounceJobGetMosaicImage =
-                        lifecycleScope.launch(Dispatchers.Main) {
-                            delay(debounceDelay)
-                            viewModel.getMosaicImage()
-                        }
+                    viewModel.emitMosaicEvent()
                 }
             })
         }
+    }
+
+    private fun observeState() = lifecycleScope.launch {
```

Comment:
현재 상황에서 반환을 하지 않아도 정상적으로 작동 합니다! Job을 반환하려는 이유가 따로 있을까요?
```suggestion
    private fun observeState() { 
        lifecycleScope.launch {
    ...
    }
```